src/base/notifications/dbusnotifier.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2015 Eugene Shalygin
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "dbusnotifier.h"
+
+#include "notifications.h"
+#include "request.h"
+
+#include <QDBusConnection>
+#include <cassert>
+
+namespace
+{
+    const QLatin1String DBUS_SERVICE_NAME("org.freedesktop.Notifications");
+    const QLatin1String DBUS_INTERFACE_NAME("org.freedesktop.Notifications");
+    const QLatin1String DBUS_PATH("/org/freedesktop/Notifications");
+    const QLatin1String APPLICATION_NAME("qBittorrent");
+    const QLatin1String APPLICATION_ICON_NAM("qbittorrent");
+}
+
+Notifications::DBusNotifier::DBusNotifier(QObject *parent)
```

Comment:
It's just the "principle of relative names". It is typically used for short.
Or do you like such long ugly signatures? It's like that you call your family members by full name (in your home).